src/django/pfb_analysis/models.py

```diff
@@ -150,6 +242,16 @@ def logs_url(self):
                           batch_job_name=self.batch_job_name,
                           batch_job_id=self.batch_job_id)
 
+    def cancel(self, reason=None):
+        """ Cancel the analysis job, if its running """
+        if not reason:
+            reason = 'AnalysisJob terminated by user at {}'.format(datetime.utcnow())
+
+        if self.status in self.Status.ACTIVE_STATUSES:
+            client = boto3.client('batch')
+            client.terminate_job(jobId=self.batch_job_id, reason=reason)
```

Comment:
`ACTIVE_STATUSES` includes `CREATED`, so trying to terminate the remote job should either be protected by `if self.batch_job_id is not None` or wrapped in a `try: ... except botocore.exceptions.ParamValidationError: ...`